src/components/Table/CollapseToggle.tsx

```diff
@@ -8,9 +8,14 @@ export interface GridTableCollapseToggleProps {
 export function CollapseToggle(props: GridTableCollapseToggleProps) {
   const { row } = props;
   const { isCollapsed, toggleCollapsed } = useContext(GridCollapseContext);
-  const toggleOnClick = useCallback(() => toggleCollapsed(row.id), [row.id, toggleCollapsed]);
-  const iconKey = isCollapsed(row.id) ? "chevronRight" : "chevronDown";
-  const headerIconKey = isCollapsed(row.id) ? "chevronsRight" : "chevronsDown";
+  const [, setTick] = useState(0);
```

Comment:
I'm a little worried that a row could be doing other custom `isCollapsed` logic and will not get re-rendered when just this local `setTick` is called.

I liked the `collapseAllContext` vs. `collapseRowContext` change, but is there a tldr of why we couldn't keep using `isCollapsed=true/false` as a boolean prop on `GridRow` and let that drive re-rendering of the whole row?

Reply:
We could probably go back to the old way of doing things. I think the main win here is to have the GridRow as a child of the CollapseContext, such that changes to the toggleCollapsed or isCollapsed do not trigger a rerender of the whole row, but just the toggle caret.

This change has the effect of:

Toggling all rows will rerender all toggle carets
Adding/removing rows will not trigger rerenders (except for the header row)

If we went back to a `isCollapsed: boolean`, then we would end up with a new context.value every time, which would always rerender all the carets. But maybe we shouldn't care, there aren't so many of them 🤷

Reply:
Personally I'm fine with row-level re-renders on collapse state change, as I think a thesis of the virtualized row approach is that ~20-30 rows at a time should still render very quickly. And, per original comment, I think eventually it'd be a feature that non-`ToggleCollapse` cells could observe the current state.

But I'm happy to defer to you on this PR; especially if the "did / did not render" behavior you want is covered by unit tests so that we can futz with things weeks/months down the road and not have regressions.

Reply:
Yea, given the perf I've seen (and setting lofty goals), I'm trying to have really fine grained rerenders 🤷 

I've [spun up another PR](https://github.com/homebound-team/beam/pull/410/files) to work on testing the rerender behaviour. Could we run this one through as-is?